agent/checker.go

```diff
@@ -4,31 +4,61 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"log"
 	"strconv"
+	"strings"
 
 	"github.com/aquasecurity/bench-common/check"
 	"github.com/pkg/errors"
 )
 
 type Checker func() (CheckResult, error)
 
-func NewChecker(definitionsPath string) (Checker, error) {
-	data, err := ioutil.ReadFile(definitionsPath)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not read definitions file")
+/* Remove
+  '---
+   controls:
+	version: ...
+	id: ...
+	description: ...
+	type: ...
+	groups:
+'
+that comes before '    - id:...'
+You need to do it to append the yaml to another yaml file
+*/
+func trim_yaml_header(yaml_text string) string {
+	pattern := "\ngroups:"
+	index := strings.Index(yaml_text, pattern)
+	if index == -1 {
+		log.Fatal("the yaml file doesn't contain the 'id:' field and is possibly broken")
 	}
+	index += len(pattern)
+	return yaml_text[index:]
+}
 
+func NewChecker(definitionsPaths []string) (Checker, error) {
+	var data [][]byte
+	for _, path := range definitionsPaths {
+		datum, err := ioutil.ReadFile(path)
```

Comment:
`datum` is not a perfect name, can we use something more descriptive?

Reply:
Done.